client/src/components/payment/payment-summary.vue

```diff
@@ -1,70 +1,53 @@
-        <small><small>Receipt No.</small></small> {{receipt.receiptNumber}}
-      </span>
-    </header>
-
-    <div v-show="fetchError">
-      <v-alert color="error" icon="mdi-alert" outlined>{{fetchError}}</v-alert>
-    </div>
-
-    <ul class="fee-list" v-show="!fetchError">
-      <li class="container fee-list__item" v-if="receipt">
-        <div class="fee-list__item-name">Amount</div>
-        <div class="fee-list__item-value">${{receipt.receiptAmount.toFixed(2)}} CAD</div>
-      </li>
-      <li class="container fee-list__item" v-if="summary">
-        <div class="fee-list__item-name">Status</div>
-        <div class="fee-list__item-value">{{summary.statusCode}}</div>
-      </li>
-      <li class="container fee-list__item">
-        <div class="fee-list__item-name">Receipt</div>
-        <div class="fee-list__item-value">
-          <v-btn text small class="primary download-receipt-btn"
-            :loading="loading" @click="downloadReceipt()">Download PDF</v-btn>
+  <v-expand-transition>
+    <!-- do not display until payments are fetched -->
+    <section class="payment-summary" v-if="summary">
+      <v-row class="py-5" no-gutters>
+        <div class="col1 align-self-center">{{receiptNumber}}</div>
+        <div class="col2 align-self-center">{{receiptDate}}</div>
+        <div class="col3 align-self-center">{{receiptDescription}}</div>
+        <div class="col4 align-self-center">${{receiptAmount}}</div>
+        <div class="col5 align-self-center">
+          <v-btn class="download-receipt-btn float-right" :loading="loading"
+            @click="downloadReceipt()">Download PDF</v-btn>
         </div>
-      </li>
-    </ul>
-
-  </v-card>
+      </v-row>
+    </section>
+  </v-expand-transition>
 </template>
 
 <script lang="ts">
 import { Component, Mixins, Prop } from 'vue-property-decorator'
-import '../../plugins/vuetify'
-
-import { NameRequestPaymentResponse } from '@/modules/payment/models'
-
-import RequestDetails from '@/components/common/request-details.vue'
 import PaymentMixin from '@/components/payment/payment-mixin'
+import ReceiptMixin from '@/components/mixins/receipt-mixin'
 
-@Component({
-  components: {
-    RequestDetails
-  }
-})
-export default class PaymentSummary extends Mixins(PaymentMixin) {
-  @Prop(Number)
-  readonly id: number
-
+@Component({})
+export default class PaymentSummary extends Mixins(PaymentMixin, ReceiptMixin) {
   @Prop(Object)
   readonly summary: any
 
-  @Prop(Object)
-  readonly receipt: any
+  /** Used to show loading state on button. */
+  private loading = false
 
-  protected fetchError = ''
+  private get receiptNumber (): string {
+    return `Receipt No. ${this.summary?.receipt.receiptNumber}`
+  }
 
-  private loading = false
+  private get receiptDate (): string {
+    return this.summary?.receipt.receiptDate
+  }
+
+  private get receiptDescription (): string {
+    const lineItem = this.summary?.payment.sbcPayment.lineItems[0] // just look at first one
```

Comment:
Should we optional chain `payment` and `sbcPayment` as well?

Reply:
"Summary" needs to be check for null/undefined because this modal is created when the app starts (not just when we ask to display it).

I don't think there's much to be gained in checking "payment" or sbcPayment. The previous code did not check it. Lots of other code will fail if those are null/undefined. I think it's OK to assume those objects are present.